sklearn/preprocessing/label.py

```diff
@@ -33,18 +34,47 @@
 ]
 
 
-def _encode_numpy(values, uniques=None, encode=False, check_unknown=True):
+def _nan_unique(ar, return_inverse=False, allow_nan=False):
+    # mimic np.unique with allow_nan option
```

Comment:
Please note that nan is removed from the unique values

Reply:
Actually, in _nan_unique it's keeping only one nan. 
Contrary to np.unique which keep them all (since there are all different by definition) .